source/common/http/async_client_impl.cc

```diff
@@ -300,7 +300,12 @@ AsyncRequestSharedImpl::AsyncRequestSharedImpl(AsyncClientImpl& parent,
 
 void AsyncRequestImpl::initialize() {
   Tracing::HttpTraceContext trace_context(request_->headers());
-  child_span_->injectContext(trace_context, nullptr);
+  Tracing::UpstreamContext upstream_context(nullptr,                    // host_
+                                            parent_.cluster_.get(),     // cluster_
+                                            Tracing::ServiceType::Http, // service_type_
```

Comment:
Right, and I don't think this code path will be executed since `AsyncRequestImpl:initialize()` is overrode in Envoy grpc although the `stream_` is from http async client.